recipe/build.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+if [ -d ${PREFIX}/include ];
+then
+    echo "Directory ${PREFIX}/include exists"
+else
+    echo "Error: Directory does not exists"
+    mkdir ${PREFIX}/include
+fi
+
+cp ${SRC_DIR}/sse2neon.h ${PREFIX}/include
+
+make check
```

Comment:
I don't think there is much value right now to run the tests, considering there is no dependencies, it's a header only library and they use `-I.`.

Reply:
yes, I thought it too, nevertheless I run tests it turns out it does not work with -02 and gcc 11, see
https://anaconda.slack.com/archives/C02K52E033M/p1700468170036429

Reply:
Yep I saw that. But it doesn't mean that it's necessarily broken with our compiler version and compiler flags. Unless we take a deeper look at the issue, we can't know if the problem stems from the the library itself or the tests. And this is really what has to be verified. If the issue is with the headers, then we have a problem, cause we'll be forced to compile downstream projects with an older compiler.

Reply:
So there is 4 tests failing with gcc 11.2:
* mm_loadu_si32
* mm_loadu_si32
* mm_get_denormals_zero_mode
* mm_set_denormals_zero_mode

`mm_loadu_si32` and `mm_loadu_si32` were added to gcc 11 but were buggy (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=99754). They were fixed in gcc 11.3 (I tested with conda-forge to confirm that it is really the case). We have 11.2 on defaults.

When a compiler doesn't know what to do with an intrisic, the library just skips the test. So if we use a gcc version older than 11, `mm_loadu_si32` and `mm_loadu_si32` are marked as skipped. Which "fixes" the problem.

`mm_get_denormals_zero_mode` and `mm_set_denormals_zero_mode` fail for an unknown reason. I'd have to attach a debugger, and even then, I don't know much about intrinsics, so i'd be hard to debug.

I think I would recommend to go with our default compiler, remove the patch, use `unset CFLAGS && unset CXXFLAGS` just before invoking `make check` and test downstream projects to see if they work correctly.

Note that I also searched online to see what other package ecosystems do and didn't see anything special about gcc 11.

Reply:
yeah, tried with unset CFLAGS & CXXFLAGS only optimization level makes a difference

Reply:
with default compiler sse2neon fails on linux64 https://cloud.prefect.io/anaconda/flow-run/772f193f-b225-4315-9cdb-b26e3571454f?logs, I got back to older gcc, but only on linux64